main.tf

```diff
@@ -332,6 +346,21 @@ resource "aws_route53_record" "aaaa_record" {
     zone_id                = aws_alb.alb.zone_id
   }
 }
+resource "aws_route53_record" "new_cert_validation" {
+  for_each = local.create_new_https_cert ? { # if https cert is not provided, then create validation records
+    for dvo in aws_acm_certificate.new_cert[0].domain_validation_options : dvo.domain_name => {
```

Comment:
Does this actually work? I feel like terraform wants to know exactly how many resources it is creating when it does a plan.

Reply:
I can get a plan to work, but it confuses me that it does work for: https://developer.hashicorp.com/terraform/language/meta-arguments/for_each#using-expressions-in-for_each. It seems like terraform wouldn't know up front what it is trying to create.

Reply:
So I'd like to actually try deploying something to make sure it works.

Reply:
Actually, I see this is how it works elsewhere, so I guess I'm crazy.

Reply:
I was able to deploy a test svelte-frontend with this, both with and without a custom url.